maintainer/deploy_docs.sh

```diff
@@ -16,27 +26,28 @@ function die () {
     exit $err
 }
 
-DOCDIR="package/doc/html"
-REPOSITORY="github.com/MDAnalysis/mdanalysis.git"
-
 rev=$(git rev-parse --short HEAD)
 
+# the following tests should be superfluous because of -o nounset
 test -n "${GH_TOKEN}" || die "GH_TOKEN is empty: need OAuth GitHub token to continue" 100
-cd $DOCDIR || die "Failed to 'cd $DOCDIR'. Run from the top level of the repository"
+test -n "${GH_REPOSITORY}" || die "GH_REPOSITORY must be set in .travis.yml" 100
+test -n "${MDA_DOCDIR}" || die "MDA_DOCDIR must be set in .travis.yml" 100
+
+cd ${MDA_DOCDIR} || die "Failed to 'cd ${MDA_DOCDIR}'. Run from the top level of the repository"
 
 git init
-git config user.name "Travis CI"
-git config user.email "TravisCI@mdanalysis.org"
+git config user.name "${GIT_CI_USER}"
+git config user.email "${GIT_CI_EMAIL}"
 
-git remote add upstream "https://${GH_TOKEN}@${REPOSITORY}"
-git fetch upstream
+git remote add upstream "https://${GH_TOKEN}@${GH_REPOSITORY}"
+git fetch --depth 50 upstream ${GH_DOC_BRANCH} gh-pages
```

Comment:
I suppose we could reduce depth, maybe even `--depth=1`... but I got fed up with testing.